pkg/vsphere/compute/rp.go

```diff
@@ -88,3 +88,51 @@ func (rp *ResourcePool) GetCluster(ctx context.Context) (*object.ComputeResource
 
 	return object.NewComputeResource(rp.Client.Client, mrp.Owner), nil
 }
+
+func (rp *ResourcePool) GetDatacenter(ctx context.Context) (*object.Datacenter, error) {
+	dcRef, err := rp.getLowestAncestor(ctx, "Datacenter")
+
+	if err != nil || dcRef == nil {
+		log.Errorf("Unable to get datacenter ancestor of rp %s: %s", rp.Name(), err)
+		return nil, err
+	}
+
+	return object.NewDatacenter(rp.Client.Client, *dcRef), nil
+}
+
+func (rp *ResourcePool) getAncestors(ctx context.Context, inType string) ([]types.ManagedObjectReference, error) {
+	client := rp.Session.Vim25()
+
+	ancestors, err := mo.Ancestors(ctx, client, client.ServiceContent.PropertyCollector, rp.Reference())
+
+	if err != nil {
+		log.Errorf("Unable to get ancestors of rp %s: %s", rp.Name(), err)
+		return nil, err
+	}
+
+	outAncestors := make([]types.ManagedObjectReference, 0)
+	for _, ancestor := range ancestors {
+		if ancestor.Self.Type == inType {
+			a := ancestor.Self
+			outAncestors = append(outAncestors, a)
+		}
+	}
+
+	return outAncestors, nil
+}
+
+func (rp *ResourcePool) getLowestAncestor(ctx context.Context, inType string) (*types.ManagedObjectReference, error) {
+	ancestors, err := rp.getAncestors(ctx, inType)
+
```

Comment:
Unneeded whitespace.

Reply:
Is the unneeded whitespace between the two return values? Go fmt works on this set of changes. I would be wary of removing whitespaces if go fmt doesn't complain.

Reply:
There's unneeded whitespace between calling a function that returns an error and the `if` condition that checks for the error.

Reply:
Ah, ok you mean there is an extra/unnecessary newline in between the function call and the testing of the condition. Ok, I will fix that. Thanks for the review!!